cpp/src/slice2cs/Gen.cpp

```diff
@@ -655,23 +722,52 @@ Slice::CsVisitor::writeDataMemberInitializers(const DataMemberList& dataMembers)
 }
 
 void
-Slice::CsVisitor::writeDocComment(const ContainedPtr& p)
+Slice::CsVisitor::writeDocComment(const ContainedPtr& p, const string& generatedType, const string& notes)
 {
     optional<DocComment> comment = DocComment::parseFrom(p, csLinkFormatter, true, true);
-    if (!comment)
+    if (comment)
     {
-        return;
+        writeDocLines(_out, "summary", comment->overview());
     }
 
-    StringList overview = comment->overview();
-    if (!overview.empty())
+    // We generate remarks only for module-level types.
+    if (dynamic_pointer_cast<Module>(p->container()))
     {
```

Comment:
Might be nice to add an `assert(!generatedType.empty());` here, since if we have a module-level type, it _has_ to provide this field to avoid generated a poorly formatted `remarks` section.

Reply:
Fixed as suggested.